### PR TITLE
Implement new design of register a delegate page

### PR DESCRIPTION
**Pull request by osvaldovega**
### What issue have I solved?

-- #2082 

### How have I implemented/fixed it?

This PR includes the new design for delegate registration.
nickname validation (is already exist or if has valid characters or not).
If user has enough funds or if user is a delegate already.

In the summary page show the fee amount of the tx with the selected nickname and user address.

In the result page or final page after submit the transactions will display the animation based on the result of the tx, in case of success will display a button that will redirect to delegates page, but in case of error a link for report the error and a retry button to submit the tx again.

### How has this been tested?

1. login to the app with an account that is not a delegate already.
2. then go to delegates tab
3. do a click in Become a delegate button
4. type the nickname that you want (you can type anything even special characters and see the validations)
5. Do a click in the confirmation button
6. Check the summary page.
7. Then submit the transactions, (if you want you can click in the in the Go Back button and see the previous page agin)
8. After submit the transaction you will see the animation and once tx is confirmed you will see the success status or not if fail.

**NOTE:** If you select and account with not enough funds you wont be able to do the process.

### Review checklist
- The PR follows our [Test guide](/LiskHQ/lisk-hub/blob/development/docs/TEST_GUIDE.md)
- The PR follows our [CSS guide](/LiskHQ/lisk-hub/blob/development/docs/CSS_GUIDE.md)

### CODE REVIEW

File: src/components/delegateRegistration/delegateRegistration.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+  // istanbul ignore next
+  goBackToDelegates() {
+    this.props.history.push(routes.delegates.path);
+  }
+
+  // istanbul ignore next
+  submitDelegateRegistration(userInformation) {
+    this.props.delegateRegistered(userInformation);
+  }
```
Comment (slaweet) [inline src/components/delegateRegistration/delegateRegistration.js]: This method can IMO be removed and use directly `props.delegateRegistered` instead of it. 

It's IMO more readable to have the same name for the action (`delegateRegistered`) in this component and in its children.

File: src/components/delegateRegistration/delegateeRegistration.test.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  });
+
+  it('renders properly SelectName component', () => {
+    expect(wrapper).toContainMatchingElement('.select-name-container');
+    expect(wrapper).toContainMatchingElements(2, '.select-name-text-description');
+    expect(wrapper).toContainMatchingElement('.learm-more-link');
+    expect(wrapper).toContainMatchingElement('.select-name-input');
+    expect(wrapper).toContainMatchingElement('.input-feedback');
+    expect(wrapper).toContainMatchingElement('.confirm-btn');
+  });
```
Comment (slaweet) [inline src/components/delegateRegistration/delegateeRegistration.test.js]: Here I would expect some tests that will
- click through the whole process (with success and failure)
- check the edge cases (nickname taken, low balance, invalid name)

File: src/components/delegateRegistration/selectName/SelectName.js [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    const { t } = this.props;
+    if (nickname.length > 20) return t('Nickname is too long.');
+    const hasInvalidChars = nickname.replace(regex.delegateSpecialChars, '');
+    if (hasInvalidChars) return t(`Invalid character ${hasInvalidChars.trim()}`);
+    return false;
+  }
+
+  onUserFetch(nickname, error) {
+    if (!error && nickname.length) {
+      this.props.delegatesFetched({ username: nickname });
```
Comment (slaweet) [inline src/components/delegateRegistration/selectName/SelectName.js]: I think this is one of the cases where we don't need to use redux and can have the API request directly in the component because the data from the API is not used by any other component.

File: src/components/delegateRegistration/selectName/SelectName.js [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+              value={nickname}
+              placeholder={t('ie. Peter Pan')}
+              className={`${styles.inputNickname} select-name-input`}
+              disabled={inputDisabled}
+            />
+            <SpinnerV2 className={`${styles.spinner} ${loading && nickname.length ? styles.show : styles.hide} spiner`}/>
+            <Icon
+              className={`${styles.status} ${!loading && nickname.length ? styles.show : styles.hide} input-status-icon`}
+              name={error ? 'alert_icon' : 'ok_icon'}
+            />
```
Comment (slaweet) [inline src/components/delegateRegistration/selectName/SelectName.js]: Plese put a TODO comment here
```
// TODO <Spiner/> and <Icon/> will be incorporated into <InputV2/> in https://github.com/LiskHQ/lisk-hub/issues/2091
```

File: src/components/delegateRegistration/status/status.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    this.setState({ status: 'pending' });
+    submitDelegateRegistration(userInfo);
+  }
+
+  render() {
+    const { t, goBackToDelegates } = this.props;
+    const { status } = this.state;
+
+    const isTransactionSuccess = status !== 'fail';
+    // istanbul ignore next
```
Comment (slaweet) [inline src/components/delegateRegistration/status/status.js]: Can be removed after resolving my comment in `src/components/delegateRegistration/delegateeRegistration.test.js`

File: test/cypress/utils/enterSecondPassphrase.js [hunk 6]
```diff
@@ -5,6 +5,14 @@ const secondPassphraseSS = {
   secondPassphraseNextBtn: '.second-passphrase-next',
 };
 
+export const enterPassphrase = (passphrase) => {
```
Comment (slaweet) [inline test/cypress/utils/enterSecondPassphrase.js]: This is duplicate to `export function enterSecondPassphraseV2(passphrase) {` down in this file. 
Now, the original `export default function enterSecondPassphrase` is IMO not used anywhere, so `enterSecondPassphraseV2` can be renamed to `enterSecondPassphrase` and replace it

File: src/components/delegateRegistration/status/status.js [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+  }
+
+  // istanbul ignore next
+  onCheckTransactionStatus() {
+    const { status } = this.state;
+    const { delegate } = this.props;
+
+    if (delegate.registerStep && delegate.registerStep === 'register-success' && status === 'pending') {
+      this.setState({ status: 'ok' });
+    }
```
[the remark this answers is not in the document]
Comment (osvaldovega) [reply src/components/delegateRegistration/status/status.js]: yes this is going to be addressed in PR https://github.com/LiskHQ/lisk-hub/pull/2186

File: src/components/delegateRegistration/selectName/SelectName.js (hunk 3 above)
Comment (osvaldovega) [reply src/components/delegateRegistration/selectName/SelectName.js]: yes this is going to be addressed in PR https://github.com/LiskHQ/lisk-hub/pull/2186

File: src/components/delegateRegistration/status/status.js (hunk 5 above)
Comment (osvaldovega) [reply src/components/delegateRegistration/status/status.js]: yes this is going to be addressed in PR https://github.com/LiskHQ/lisk-hub/pull/2186

**Comment by osvaldovega** [comment]
Thx @slaweet I fixed all the issues mentioned above.
There are a couple of comments as you already know that will be fix/implement on PR #2186.
In the same PR #2186 I will update the delegateRegistration file for full coverage because the logic for request the information is going to change I prefer to do it in that PR after the change

**Comment by reyraa** [comment]
@osvaldovega The input has a placeholder : Peter Pan. but this is an invalid value (shows an error). Please replace it with a valid placeholder.

**Comment by slaweet** [comment]
As discussed in person:
- :bug: still no red outline
- :fire: `test/cypress/utils/enterSecondPassphrase.js`needs more clean up
- :recycle: `goBackToDelegates={history.push}` in `src/components/delegateRegistration/delegateRegistration.js` is confusing

**Comment by slaweet** [comment]
Functionality and code are good now. 👍 

Just e2e test failing on Jenkins.

File: Jenkinsfile [hunk 8]
```diff
@@ -136,6 +136,7 @@ EOF
 										  echo $ret >.cypress_status
 										else
 										  FAILED_TESTS="$( awk '/Spec/{f=1}f' cypress.log |grep --only-matching '✖ .*.spec.js' |awk '{ print "test/cypress/e2e/"$2 }' |xargs| tr -s ' ' ',' )"
+										  make coldstart
```
Comment (fchavant) [inline Jenkinsfile]:
```suggestion
										  ( cd cd $WORKSPACE/$BRANCH_NAME; make coldstart )
```

File: Jenkinsfile [hunk 9]
```diff
@@ -136,6 +137,11 @@ EOF
 										  echo $ret >.cypress_status
 										else
 										  FAILED_TESTS="$( awk '/Spec/{f=1}f' cypress.log |grep --only-matching '✖ .*.spec.js' |awk '{ print "test/cypress/e2e/"$2 }' |xargs| tr -s ' ' ',' )"
+										  cd $WORKSPACE/$BRANCH_NAME
++										  make coldstart
++										  export CYPRESS_coreUrl=http://127.0.0.1:$( docker-compose port lisk 4000 |cut -d ":" -f 2 )
++										  sleep 10
++										  cd -
```
Comment (slaweet) [inline Jenkinsfile]: @Efefefef This is great. I was thinking about something like this since some time :-) 

But it looks like copy-paste from a diff (with `+` at the beginning of line)

Comment (Efefefef) [reply Jenkinsfile]: my bad

**Comment by slaweet** [comment]
I talked with @yasharAyari and we concluded the following.

> Now the reason of e2e failure fixed time to fix few small things
> 🐛 If you retry failed transaction and success you still get the transaction error screen
This will be fixed in https://github.com/LiskHQ/lisk-hub/pull/2186 as this logic is changed in that PR a lot

> 🐛 Wrong order : Fee and passphase; Show passphrase checkbox should be an eye icon
> image
> 🐛 Wrong margin
> image

These two are also same in other transaction types, so should be fixed as a separate ticket

**Comment by Efefefef** [comment]
@slaweet Which two exactly? Should I create new tickets?

**Comment by slaweet** [comment]
@Efefefef 
> 🐛 Wrong order : Fee and passphase; Show passphrase checkbox should be an eye icon
> 🐛 Wrong margin

yes, please create a ticket "Align the design of 2nd passphrase inputs in all transaction types"

**Comment by Efefefef** [comment]
So @osvaldovega for you only this 1 to fix
🐛 If you retry failed transaction and success you still get the transaction error screen

**Comment by osvaldovega** [comment]
🐛 If you retry failed transaction and success you still get the transaction error screen
This issue is fixed in PR #2186

**Comment by Efefefef** [comment]
Ok. Approve
